src/page/index.js

```diff
@@ -74,92 +89,88 @@ const api = new Api({
   },
 });
 
-api
-  .getAppInfo()
-  .then(
-    api.getUserInfo().then((res) => {
-      //assign new value for userId
-      userId = res._id;
-      //assign avatar
-      profileAvatar.src = res.avatar;
-      //new UserInfo
-      userInfo = new UserInfo({
-        title: ".profile__title",
-        subtitle: ".profile__subtitle",
-      });
-      //setting userInfo from server
-      userInfo.setUserInfo(res);
-    })
-  )
-  .then(
-    // loaded cards from server
-    api.getInitialCards().then((res) => {
-      //new section
-      cardSection = new Section(
-        {
-          //using data from server
-          data: res,
-          renderer: (item) => {
-            renderCard(item);
-          },
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, initialCards]) => {
+    userId = userData._id;
+    userInfo = new UserInfo({
+      title: ".profile__title",
+      subtitle: ".profile__subtitle",
+      avatar: ".profile__picture",
+    });
+    userInfo.setUserInfo(userData);
+    //new Section
+    cardSection = new Section(
+      {
+        //using data from server
+        data: initialCards,
+        renderer: (item) => {
+          renderCard(item);
         },
-        config.cardSectionClass
-      );
-      cardSection.renderItems();
-    })
-  )
+      },
+      config.cardSectionClass
+    );
+    cardSection.renderItems();
+  })
   .catch((err) => {
     console.error(err);
   });
 
 const popupEditForm = new PopupWithForm(
   "#profile__edit-modal",
   (inputValues) => {
-    popupEditForm.toggleSaveBtn();
-    userInfo.setUserInfo(inputValues);
+    popupEditForm.renderLoading(true);
     //updating userInfo (on?at?in) server
     api
       .updateUserInfo(inputValues)
+      .then((res) => {
+        userInfo.setUserInfo(res);
+      })
+      .then(() => {
+        popupEditForm.close();
+      })
       .catch((err) => {
         console.error(err);
       })
       .finally(() => {
-        popupEditForm.close();
-        popupEditForm.toggleSaveBtn();
+        popupEditForm.renderLoading(false);
       });
   }
 );
 
 const popupAvatarFrom = new PopupWithForm(
   "#profile__avatar_modal",
   (inputValues) => {
-    popupAvatarFrom.toggleSaveBtn();
+    popupAvatarFrom.renderLoading(true);
     api
       .updateProfilePicture(inputValues)
       .then((res) => {
         profileAvatar.src = res.avatar;
```

Comment:
`        profileAvatar.src = res.avatar;
`

Class `UserInfo` should set the entire user’s data (including the **avatar**)